File: jna/src/main/java/dark/detection.java
package dark;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;

import jdk.internal.vm.compiler.word.PointerBase;

import java.util.Arrays;
import java.util.List;

public class detection extends Structure   {
        /*
        typedef struct detection{
            box bbox;
            int classes;
            float *prob;
            float *mask;
            float objectness;
            int sort_class;
            float *uc; // Gaussian_YOLOv3 - tx,ty,tw,th uncertainty
            int points; // bit-0 - center, bit-1 - top-left-corner, bit-2 - bottom-right-corner
            float *embeddings;  // embeddings for tracking
            int embedding_size;
            float sim;
            int track_id;
        } detection;
        */

        public box.ByValue bbox;
        public int classes;
        public Pointer prob; //float*
        public Pointer mask; //float*
        public float objectness;
        public int sort_class;
        public Pointer uc; //float*
        public int points;
        public Pointer embeddings; //float*
        public int embedding_size;
        public float sim;
        public int track_id;

    public detection() {
        super();
    }

    public detection(Pointer peer) {
        // cannot use super(p) because of fixed-size array fields
        super();
        useMemory(peer); // set pointer
        read(); // initialize fields
    }

    public detection(box bbox,int classes){

    }


    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("bbox", "classes", "prob", "mask", "objectness", "sort_class", "uc",
        "points", "embeddings", "embedding_size", "sim", "track_id");
    }


    public static class ByReference extends detection implements Structure.ByReference {
        public ByReference() { }
        public ByReference(Pointer p) { super(p); read(); }
    }

}

File: jna/src/main/java/dark_kotlin/detection.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure


open class detection : Structure {
        /*
        typedef struct detection{
            box bbox;
            int classes;
            float *prob;
            float *mask;
            float objectness;
            int sort_class;
            float *uc; // Gaussian_YOLOv3 - tx,ty,tw,th uncertainty
            int points; // bit-0 - center, bit-1 - top-left-corner, bit-2 - bottom-right-corner
            float *embeddings;  // embeddings for tracking
            int embedding_size;
            float sim;
            int track_id;
        } detection;
        */

        @JvmField var bbox: box.ByValue? = null
        @JvmField var classes = 0
        @JvmField var prob: Pointer? = null //float*
        @JvmField var mask: Pointer? = null //float*
        @JvmField var objectness = 0f
        @JvmField var sort_class = 0
        @JvmField var uc: Pointer? = null //float*
        @JvmField var points = 0
        @JvmField var embeddings: Pointer? = null //float*
        @JvmField var embedding_size = 0
        @JvmField var sim = 0f
        @JvmField var track_id = 0

        constructor(): super()
        constructor(p: Pointer) : super(){
            useMemory(p)
            read()
        }

        override fun getFieldOrder(): List<String> {
            return listOf("bbox", "classes", "prob", "mask", "objectness", "sort_class", "uc",
                "points", "embeddings", "embedding_size", "sim", "track_id")
        }

        class ByReference : detection, Structure.ByReference {
            constructor() {}
            constructor(p: Pointer) : super(p) {
                read()
            }
        }
    }
